### PR TITLE
Api

### PR DESCRIPTION
**Pull request by stevenweaver**

### CODE REVIEW

File: src/jsx/apikey.jsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    this.div.appendChild(script);
+  }
+
+  recaptchaLoaded() {
+    console.log("Capcha successfully loaded");
+  }
+
+  handleSubscribe() {
+    if (this.state.isVerified) {
+      fetch("http://dev.datamonkey.org/api/v1/issueKey", {
```
Comment (stevenweaver) [inline src/jsx/apikey.jsx:57]: Hardcoded URL

File: src/jsx/apikeyinfo.jsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+  }
+
+  componentDidMount() {
+    var api_key_id = url.parse(window.location.href).pathname.substr(11);
+
+    this.setState({
+      api_key: api_key_id,
+    });
+
+    fetch("http://dev.datamonkey.org/api/v1/keyinfo", {
```
Comment (stevenweaver) [inline src/jsx/apikeyinfo.jsx:41]: Hardcoded URL

File: testFile [hunk 3]
```diff
@@ -1,46 +0,0 @@
-
```
Comment (stevenweaver) [inline testFile:1]: No file extension

File: config/setup.js.tpl [hunk 4]
```diff
@@ -1,30 +0,0 @@
-function define(name, value) {
```
Comment (stevenweaver) [inline config/setup.js.tpl:1]: Why deleted?

File: config/routes.js [hunk 5]
```diff
@@ -264,7 +262,14 @@ module.exports = function (app) {
...
-  //app.post("/api/v1/submit", api.apiSubmit);
-  app.post("/api/v1/submit", api.apiSubmit);
-  app.post("/api/v1/debug", slac.invokeDEBUG);
+  app.post("/api/v1/submit", api.checkAPIKey, api.apiSubmit);
+  app.get("/api/v1/status", api.apiStatus);
+
+  // API KEY ROUTES
+  api_verify = require(path.join(__dirname, "../app/routes/api"));
+  app.get("/apikey", api_verify.renderApi);
+  app.post("/api/v1/issueKey", api.checkCapcha, api.issueKey);
```
Comment (stevenweaver) [inline config/routes.js:271]: Can't there be a way to bypass checkCapcha for dev purposes?

File: app/templates/api.ejs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<%- include("includes/header.ejs") %>
```
Comment (stevenweaver) [inline app/templates/api.ejs:1]: Seems like these template files could be generalized into just one.

File: app/routes/api.js [hunk 7]
```diff
@@ -12,38 +9,39 @@ const mongoose = require("mongoose"),
...
 
   shortid.characters(
     "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-_"
   );
 
-  let postdata = req.body;
-
-  var url_fasta = postdata.fastaLoc,
-    fileName = "api_" + shortid.generate() + "z.nex",
+  var website_url = "datamonkey.org", //Used to build reply URL
```
Comment (stevenweaver) [inline app/routes/api.js:32]: Hardcoded URL

File: app/routes/api.js [hunk 8]
```diff
@@ -12,38 +9,39 @@ const mongoose = require("mongoose"),
...
-  let datatype = 0,
-    gencodeid = postdata.gencodeid;
-
-  let options = {
-    datatype: 0,
-    gencodeid: gencodeid,
-    mail: postdata.mail,
-  };
-
   function getRequest(url, dest, callback) {
```
Comment (stevenweaver) [inline app/routes/api.js:45]: No filesize limit.

File: config/setup.js.tpl (hunk 4 above)
Comment (kjlevitz) [reply config/setup.js.tpl:1]: Most likely because I renamed it. I'll address.

File: app/templates/api.ejs (hunk 6 above)
Comment (kjlevitz) [reply app/templates/api.ejs:2]: Dev mode toggle has now been added to the config file. It bypasses the API key check when making requests.

File: config/routes.js (hunk 5 above)
Comment (kjlevitz) [reply config/routes.js:271]: Dev mode toggle has now been added to the config file. It allows you to bypasses the API key check when making requests.

File: app/routes/api.js (hunk 8 above)
Comment (kjlevitz) [reply app/routes/api.js:74]: Latest update addresses this and adds a file-size limit that is configurable in setup file.

**Pull request merged**